priority_queuee.h

```diff
@@ -0,0 +1,329 @@
+
+
+/* @brief: This function is used to enqueue (push) the data to the PriorityQueue array ascendingly
+ * @para : *pQueue ::
+ * 					pointer to the stack that data will be enqueued in.
+ *
+ * 			Data  ::
+ *					The data that will be enqueue to the queue ascendingly
+ * @ret  : None
+ * @note : This function must be used every time you want to enqueue the data to the PriorityQueue array.
+ *
+ * */
+void     PriorityQueue_Enqueue    (PriorityQueue_t *pQueue , QueueEntry Data);
+
+
+
+/* @brief: This function is used to dequeue data from the queue array
+ * @para : pQueue ::
+ * 					pointer to the queue that the data will be dequeue out from it.
+ * @ret  : None
+ * @note  :This function must be used every time you want to dequeue the data from queue array.
+ *
+ * */
+void     PriorityQueue_Dequeue    (PriorityQueue_t *pQueue);
+
+
+/* @brief: This function is used to clear the queue array (delete Queue)
+ * @para : *stack ::
+ * 					pointer to the queue that you want to delet it.
+ * @ret  : None
+ * @note : None
+ *
+ * */
+void     PriorityQueue_ClearQueue (PriorityQueue_t *pQueue);
+
+
+
+
+
+/* @brief: This function is used to traverse each element in the queue
+ * @para : *pQueue ::
+ * 					pointer to the Queue that you want to traverse it
+ *
+ *		   void (*pFun)(QueueEntry Data)::
+ *					pFun is a pointer to function which take one argument of the queue data type
+ *					and return nothing
+ * @ret  : None
+ * @note : None
+ *
+ * */
+void     PriorityQueue_Traverse   (PriorityQueue_t *pQueue , void (*pFun)(QueueEntry Data));
+
+
+
+/* @brief: This function is used to check if Priority queue is empty or not
+ * @para : *pQueue ::
+ * 					pointer to the queue that you want to checkis  if it empty or not
+ *
+ * @ret  : Return 1 --> if the Queue is empty
+ *  	   Return 0 --> if the Queue is not empty
+ *
+ * @note  : None
+ *
+ * */
+uint8_t  PriorityQueue_IsEmpty    (PriorityQueue_t *pQueue);
+
+
+
+
+/* @brief: This function is used to check if Priority queue is Full or not
+ * @para : *pQueue ::
+ * 					pointer to the queue that you want to checkis  if it Full or not
+ *
+ * @ret  : Return 1 --> if the Queue is Full
+ *  	   Return 0 --> if the Queue is not Full
+ *
+ * @note  : None
+ */
+uint8_t  PriorityQueue_IsFull     (PriorityQueue_t *pQueue);
+
+
+
+/* @brief: This function is used to get the size of the Priority queue array
+ * @para : *pQueue::
+ * 					pointer to the queue that you want to know the size of it
+ *
+ *
+ * @ret  : uint16_t --> return the size of the queue
+ * @note : None
+ *
+ * */
+uint16_t PriorityQueue_GetSize    (PriorityQueue_t *pQueue);
+
+
+
+
+/* @brief: This function is used to return the highest data priority in the queue
+ * @para : *pQueue ::
+ * 					pointer to the queue that you want to return  the highest priority data in the queue
+ *
+ * @ret  : QeueuEntry --> highst data priority on the queue
+ * @note : None
+ *
+ * */
+QueueEntry PriorityQueue_GetTop (PriorityQueue_t *pQueue);
+
```

Comment:
The prototype for all types of priority queue must be same so we have not need to check for PQ type